rioxarray/_io.py

```diff
@@ -1262,6 +1262,12 @@ def open_rasterio(
 
     if chunks is not None:
         result = _prepare_dask(result, riods, filename, chunks)
```

Comment:
Should `_prepare_dask` set the `preferred_chunks` in encoding instead of the current behavior?

Reply:
My understanding is that `_prepare_dask` is called when chunks is not None and thus that the returned DataArray will contain dask arrays already.
In the case of the xarray backend, chunks should actually be None so that rioxarray return LazilyIndexedOuterArrays instead of dask arrays which is how xarray wants it (it can then do it's own chunking etc).
So in my opinion `_prepare_dask` and `preferred_chunks` are two very separate use cases, and hence I think it's better to leave `preferred_chunks` separate.

Reply:
Sounds like a good strategy - thanks for the explanation.